### DISCUSSION TITLE
Hot to get Is it possible to get information about time of the rtsp/hls stream?

### DISCUSSION THREAD

**Discussion by batrlatom** [bluenviron/mediamtx#409] [Questions]
I have an application using gstreamer ( nvidia deepstream ) which analyses the videofeed. Data from the analysis is stored in the kafka or database together with the timestamp or frameId. Output of the video stream is sent to the rtsp-simple-server as rtsp sink.

What I want to achieve is to synchronize data stored in the database together with the playback in the video stream. Is it possible to get the information about the current time in the video?

**Comment by aler9** [comment]
Hello, in RTSP every frame is associated with a unique timestamp, that is available in the header of every RTP packet. Therefore, if you want to join video frames with their associated data, you have to use this field. 

If you're using GStreamer to perform the joining, the timestamp is available in the `pts` member of the `GstBuffer` structure (https://gstreamer.freedesktop.org/documentation/gstreamer/gstbuffer.html?gi-language=c).

**Reply by aler9** [reply]
Regarding HLS, every TS packet has a PTS field too, but the problem is that it has a different initial offset than the RTSP one.

[a comment by an automated account is not shown]
